src/config/mod.rs

```diff
@@ -210,19 +214,80 @@ inventory::collect!(SourceDescription);
 pub struct SinkOuter {
     #[serde(default)]
     pub buffer: crate::buffers::BufferConfig,
-    #[serde(default = "healthcheck_default")]
-    pub healthcheck: bool,
+    // We are accepting bool and uri for backward compatibility reasons.
+    #[serde(deserialize_with = "crate::serde::bool_or_struct")]
+    #[serde(default)]
+    healthcheck: HealthcheckOptions,
+    // We are accepting this option for backward compatibility.
+    healthcheck_uri: Option<UriSerde>,
     pub inputs: Vec<String>,
     #[serde(flatten)]
     pub inner: Box<dyn SinkConfig>,
 }
 
 impl SinkOuter {
+    pub fn new(inputs: Vec<String>, inner: Box<dyn SinkConfig>) -> Self {
+        SinkOuter {
+            buffer: Default::default(),
+            healthcheck: HealthcheckOptions::default(),
+            healthcheck_uri: None,
+            inner,
+            inputs,
+        }
+    }
+
     pub fn resources(&self, name: &str) -> Vec<Resource> {
         let mut resources = self.inner.resources();
         resources.append(&mut self.buffer.resources(name));
         resources
     }
+
+    pub fn healthcheck(&self) -> HealthcheckOptions {
+        if self.healthcheck_uri.is_some() && self.healthcheck.uri.is_some() {
+            warn!("Both `healthcheck.uri` and `healthcheck_uri` options are specified. Using value of `healthcheck.uri`.")
+        } else if self.healthcheck_uri.is_some() {
+            warn!("`healthcheck_uri` option has been deprecated, use `healthcheck.uri` instead. ")
+        }
+        HealthcheckOptions {
+            uri: self
+                .healthcheck
+                .uri
+                .clone()
+                .or_else(|| self.healthcheck_uri.clone()),
+            ..self.healthcheck.clone()
+        }
+    }
+}
+
+#[derive(Deserialize, Serialize, Debug, Clone)]
+#[serde(default)]
+pub struct HealthcheckOptions {
```

Comment:
Rust appears to be complaining about multiple definitions of this, do you know what that's about?

Reply:
A, forgot about that, there was another PR that added a struct with the same name.